### PR TITLE
analyzer should run with -A for bug-report based on issue#31918

### PR DESCRIPTION
**Pull request by zhlsunshine**
This PR was submitted for issue: [issue#31918](https://github.com/istio/istio/issues/31918)

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Securitygit commit -m "this submit is for issue#31918"
[x] Test and Release
[x] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by zhlsunshine** [comment]
**Here is the test result:**

`$ istioctl bug-report`
`......
Fetching Istio control plane information from cluster.

Running istio analyze on all namespaces and report as below: 
################################################################
################################################################
Info [IST0102] (Namespace calico-system) The namespace is not enabled for Istio injection. Run 'kubectl label namespace calico-system istio-injection=enabled' to enable it, or 'kubectl label namespace calico-system istio-injection=disabled' to explicitly mark it as not needing injection.
Info [IST0102] (Namespace istio-system) The namespace is not enabled for Istio injection. Run 'kubectl label namespace istio-system istio-injection=enabled' to enable it, or 'kubectl label namespace istio-system istio-injection=disabled' to explicitly mark it as not needing injection.
Info [IST0102] (Namespace tigera-operator) The namespace is not enabled for Istio injection. Run 'kubectl label namespace tigera-operator istio-injection=enabled' to enable it, or 'kubectl label namespace tigera-operator istio-injection=disabled' to explicitly mark it as not needing injection.
Info [IST0118] (Service calico-typha.calico-system) Port name calico-typha (port: 5473, targetPort: calico-typha) doesn't follow the naming convention of Istio port.
Info [IST0118] (Service productpage.default) Port name httpd (port: 9080, targetPort: 9080) doesn't follow the naming convention of Istio port.
################################################################
################################################################
Creating an archive at /home/zhlsunshine/go/src/istio/bug-report.tar.gz.
Cleaning up temporary files in /tmp/bug-report.
Done.
......`

**Comment by zhlsunshine** [comment]
Hi @howardjohn  could you please help to do code review for this PR related to [issue#31918](https://github.com/istio/istio/issues/31918), thanks!

File: tools/bug-report/pkg/bugreport/bugreport.go [hunk 1]
```diff
@@ -407,19 +406,18 @@ func getLog(client kube.ExtendedClient, resources *cluster2.Resources, config *c
...
-		writeFiles(archive.AnalyzePath(tempDir, ns), out)
+func runAnalyze(config *config.BugReportConfig, params *content.Params) {
+	newParam := params.SetNamespace(common.NamespaceAll)
+	common.LogAndPrintf("Running istio analyze on all namespaces and report as below:")
+	out, err := content.GetAnalyze(newParam.SetIstioNamespace(config.IstioNamespace))
+	if err != nil {
+		log.Error(err.Error())
+		return
 	}
+	common.LogAndPrintf("\n################################################################\n")
```
Comment (howardjohn) [inline tools/bug-report/pkg/bugreport/bugreport.go:417]: This pattern of printing a bunch of `###` does not seem used elsewhere, do we need it?

Comment (zhlsunshine) [reply tools/bug-report/pkg/bugreport/bugreport.go:417]: @howardjohn no, I just want to split the report with some description information.

Comment (zhlsunshine) [reply tools/bug-report/pkg/bugreport/bugreport.go:417]: okay, I will remove them.

**Comment by zhlsunshine** [comment]
Hi @howardjohn , it's done, please help to check it again, thanks! ^_^

**Comment by zhlsunshine** [comment]
And **Here is the newest result:**

`$ istioctl bug-report`
`......`
`Fetching Istio control plane information from cluster.`

`Running istio analyze on all namespaces and report as below: `
`Analysis Report:`
`Info [IST0102] (Namespace calico-system) The namespace is not enabled for Istio injection. Run 'kubectl label namespace calico-system istio-injection=enabled' to enable it, or 'kubectl label namespace calico-system istio-injection=disabled' to explicitly mark it as not needing injection.`
`Info [IST0102] (Namespace istio-system) The namespace is not enabled for Istio injection. Run 'kubectl label namespace istio-system istio-injection=enabled' to enable it, or 'kubectl label namespace istio-system istio-injection=disabled' to explicitly mark it as not needing injection.`
`Info [IST0102] (Namespace tigera-operator) The namespace is not enabled for Istio injection. Run 'kubectl label namespace tigera-operator istio-injection=enabled' to enable it, or 'kubectl label namespace tigera-operator istio-injection=disabled' to explicitly mark it as not needing injection.`
`Info [IST0118] (Service calico-typha.calico-system) Port name calico-typha (port: 5473, targetPort: calico-typha) doesn't follow the naming convention of Istio port.`
`Info [IST0118] (Service productpage.default) Port name httpd (port: 9080, targetPort: 9080) doesn't follow the naming convention of Istio port.`
`Creating an archive at /home/zhlsunshine/go/src/istio/bug-report.tar.gz.`
`Cleaning up temporary files in /tmp/bug-report.`
`Done.`
`......`

**Review by howardjohn — APPROVED** [review]

**Pull request merged**